azure/scope/cluster.go

```diff
@@ -228,6 +228,11 @@ func (s *ClusterScope) PublicIPSpecs() []azure.ResourceSpecGetter {
 
 // LBSpecs returns the load balancer specs.
 func (s *ClusterScope) LBSpecs() []azure.ResourceSpecGetter {
+	apiBackendPoolName := s.APIServerLBPoolName(s.APIServerLB().Name)
+	if s.APIServerLB().BackendPool.Name != nil {
```

Comment:
Let's move this defaulting to a webhook, see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2432#discussion_r915240838